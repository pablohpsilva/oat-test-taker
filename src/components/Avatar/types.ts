export default interface IAvatar {
    alt: string
    picture: string
    size?: number
}