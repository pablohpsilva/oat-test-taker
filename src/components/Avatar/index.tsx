import React from 'react'
import styled from 'styled-components/macro'

const StyledWrapper = styled.div`
    border-radius: calc(500 * var(--s));
    ${({ size }) =>
        size &&
        `
        height: calc(${size} * var(--s));
        width: calc(${size} * var(--s));
    `}
    ${({ picture }) =>
        picture &&
        `
        background: url(${picture});
        background-position: center;
        background-size: cover;
    `}
`

const Avatar = ({ alt, picture, size = 80 }) => {
    return (
        <StyledWrapper picture={picture} size={size}>
            <picture>
                <img src={picture} alt={alt} hidden />
            </picture>
        </StyledWrapper>
    )
}

export default Avatar