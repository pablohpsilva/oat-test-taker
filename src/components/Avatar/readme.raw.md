# Avatar

## Props

| Property  | Description                 | Type     | Default        |
| --------- | --------------------------- | -------- | -------------- |
| `alt`     | Alt text for accessibility. | `string` | `undefined`    |
| `picture` | Avatar picture              | `string` | `'avatar.svg'` |
| `size`    | Avatar size                 | `number` | `undefined`    |
