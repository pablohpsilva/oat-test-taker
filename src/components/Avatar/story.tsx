import React from 'react'
import { storiesOf } from '@storybook/react'
import { text, number } from '@storybook/addon-knobs'

import { Avatar } from '@/components'
import readme from './readme.raw.md'

const AvatarComp = ({
    picture = text('picture', 'https://placeimg.com/600/300/any'),
    alt = text('alt', 'https://placeimg.com/600/300/any'),
    size = number('size', 86),
}) => <Avatar picture={picture} alt={alt} size={size} />

const opts = {
    notes: { markdown: readme }
}

storiesOf('Components|Avatar', module)
    .addDecorator(storyFn => <div>{storyFn()}</div>)
    .add('Example', () => <AvatarComp />, opts)
