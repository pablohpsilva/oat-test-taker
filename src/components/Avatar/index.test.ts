import React from 'react'
import Avatar from './'

test('should render the Avatar component', async () => {
    const wrapper = shallow(<Avatar />)

    expect(wrapper.first().hasClass('Avatar')).toMatchSnapshot()
    expect(wrapper).toMatchSnapshot()
})
