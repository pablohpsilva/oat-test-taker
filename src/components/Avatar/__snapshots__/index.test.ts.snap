// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Avatar component 1`] = `false`;

exports[`should render the Avatar component 2`] = `
<Avatar__StyledWrapper
  picture=""
  size={80}
>
  <picture>
    <img
      hidden={true}
      onError={[Function]}
      src=""
    />
  </picture>
</Avatar__StyledWrapper>
`;
